data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/consumer/JsonConsumer.java

```diff
@@ -0,0 +1,117 @@
+import org.opensearch.dataprepper.plugins.kafka.source.configuration.TopicConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * * A helper class which helps to process the JSON records and write them into
+ * the buffer. Offset handling and consumer re balance are also being handled
+ */
+
+@SuppressWarnings("deprecation")
+public class JsonConsumer implements KafkaSchemaTypeConsumer<String, JsonNode>, ConsumerRebalanceListener {
+  private static final Logger LOG = LoggerFactory.getLogger(JsonConsumer.class);
+  private Map<TopicPartition, OffsetAndMetadata> currentOffsets = new HashMap<>();
+  private KafkaConsumer<String, JsonNode> kafkaJsonConsumer;
+  private long lastReadOffset = 0L;
+  private final JsonFactory jsonFactory = new JsonFactory();
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static final String MESSAGE_KEY = "message";
+	
+  @SuppressWarnings({ "rawtypes", "unchecked" })
+  public void consumeRecords(KafkaConsumer<String, JsonNode> consumer, AtomicBoolean status,
+      Buffer<Record<Object>> buffer, TopicConfig topicConfig, PluginMetrics pluginMetrics) {
+    KafkaSourceBuffer kafkaSourceBuffer = new KafkaSourceBuffer(topicConfig, pluginMetrics);
+    kafkaJsonConsumer = consumer;
+    try {
+      currentOffsets.clear();
+      consumer.subscribe(Arrays.asList(topicConfig.getName()));
+      while (!status.get()) {
+        ConsumerRecords<String, JsonNode> records = consumer.poll(Duration.ofMillis(100));
+        if (!records.isEmpty() && records.count() > 0) {
+          for (TopicPartition partition : records.partitions()) {
+            List<ConsumerRecord<String, JsonNode>> partitionRecords = records.records(partition);
+            for (ConsumerRecord<String, JsonNode> consumerRecord : partitionRecords) {
+              currentOffsets.put(new TopicPartition(consumerRecord.topic(), consumerRecord.partition()),
+                new OffsetAndMetadata(consumerRecord.offset() + 1, null));
+              writeToBuffer(consumerRecord.value(), buffer);	
+              lastReadOffset = partitionRecords.get(partitionRecords.size() - 1).offset();
+              kafkaSourceBuffer.commitOffsets(partition, lastReadOffset, consumer);
+            }
+          }
+        }
+      }
+    } catch (Exception exp) {
+      LOG.error("Error while reading the json records from the topic...{}", exp.getMessage());
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public void writeToBuffer(final JsonNode jsonNode, Buffer<Record<Object>> buffer) throws TimeoutException {
+    try {
+      final JsonParser jsonParser = jsonFactory.createParser(jsonNode.toString());
+      final Map<String, Object> innerJson = objectMapper.readValue(jsonParser, Map.class);
+      Event event = JacksonLog.builder().withData(innerJson).build();
+      Record<Object> jsonRecord = new Record<>(event);
+      buffer.write(jsonRecord, 1200);
```

Comment:
Define 1200 as a constant variable and use it. Can this be configurable in YAML and have a default value ?